certbot-nginx/certbot_nginx/tests/util.py

```diff
@@ -34,20 +35,20 @@ def setUp(self):
             "rsa512_key.pem"))
 
     def tearDown(self):
-        # On Windows we have various files which are not correctly closed at the time of tearDown.
-        # For know, we log them until a proper file close handling is written.
-        # Useful for development only, so no warning when we are on a CI process.
-        def onerror_handler(_, path, excinfo):
-            """On error handler"""
-            if not os.environ.get('APPVEYOR'):  # pragma: no cover
-                message = ('Following error occurred when deleting path {0}'
-                           'during tearDown process: {1}'.format(path, str(excinfo)))
-                warnings.warn(message)
-
-        shutil.rmtree(self.temp_dir, onerror=onerror_handler)
-        shutil.rmtree(self.config_dir, onerror=onerror_handler)
-        shutil.rmtree(self.work_dir, onerror=onerror_handler)
-        shutil.rmtree(self.logs_dir, onerror=onerror_handler)
+        # Cleanup opened resources after a test. This is usually done through atexit handlers in
+        # Certbot, but during tests, atexit will not run registered functions before tearDown is
+        # called and instead will run them right before the entire test process exits.
+        # It is a problem on Windows, that does not accept to clean resources before closing them.
+        logging.shutdown()
+        # Remove logging handlers that have been closed so they won't be
+        # accidentally used in future tests.
+        logging.getLogger().handlers = []
```

Comment:
I get the addition of the call to `util._release_lock()` but needing to clean up loggers here surprises me. Do we definitely need to do this or did you just copy it from the `tearDown` code in Certbot's tests? If the former, do you understand why?

Reply:
Oh, I just copy the code. It is not necessary.

Reply:
Fixed that. By the way, why this class does not extend TempDirTestCase from `certbot.tests.util`?

Reply:
I didn't look through the git history to verify this but I think it's that the code here predates `TempDirTestCase` and maybe there was a bit of concern about what code in Certbot the plugins should be using in a clean design.